### PR TITLE
MINOR: Implemented control of avatar by HMD roll

### PR DESCRIPTION
**Pull request by accepttestvr**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can one of the admins verify this patch?

**Comment by themelissabrown** [comment]
Ok to test

**Comment by themelissabrown** [comment]
@accepttestvr can you resolve the failing checks and add a test plan?

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -1816,8 +1826,17 @@ void MyAvatar::updateOrientation(float deltaTime) {
...
-    // use head/HMD orientation to turn while flying
+    if (_useHMDRollControl) {
+        // Update body yaw with HMD roll angle
+        const float rollAngle = glm::degrees(safeEulerAngles(getHMDSensorOrientation()).z);
+        const float rollSign = rollAngle > 0.0f ? 1.0f : -1.0f;
+        const float absRollValue = fabsf(rollAngle);
+        float yawChangeDueRoll = absRollValue > _rollControlDeadZone ? rollSign * (absRollValue - _rollControlDeadZone) : 0.0f;
+        totalBodyYaw += yawChangeDueRoll * deltaTime * _rollControlSpeed;
+    }
+    else
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: coding standard: should be `} else if (...) {`

File: interface/src/avatar/MyAvatar.h [hunk 2]
```diff
@@ -495,6 +499,30 @@ class MyAvatar : public Avatar {
     // results are in HMD frame
     glm::mat4 deriveBodyFromHMDSensor() const;
 
+    void setUseHMDRollControl(bool value) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard: should be single line - when in header

File: interface/src/avatar/MyAvatar.h [hunk 3]
```diff
@@ -495,6 +499,30 @@ class MyAvatar : public Avatar {
...
+        _useHMDRollControl = value;
+    }
+
+    bool getUseHMDRollControl() const {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard: should be single line - when in header

File: interface/src/avatar/MyAvatar.h [hunk 4]
```diff
@@ -495,6 +499,30 @@ class MyAvatar : public Avatar {
...
+        return _useHMDRollControl;
+    }
+
+    void setRollControlDeadZone(float value) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard: should be single line - when in header

File: interface/src/avatar/MyAvatar.h [hunk 5]
```diff
@@ -495,6 +499,30 @@ class MyAvatar : public Avatar {
...
+        _rollControlDeadZone = value;
+    }
+
+    float getRollControlDeadZone() const {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard: should be single line - when in header

File: interface/src/avatar/MyAvatar.h [hunk 6]
```diff
@@ -495,6 +499,30 @@ class MyAvatar : public Avatar {
...
+        return _rollControlDeadZone;
+    }
+
+    void setRollControlSpeed(float value) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard: should be single line - when in header

File: interface/src/avatar/MyAvatar.h [hunk 7]
```diff
@@ -495,6 +499,30 @@ class MyAvatar : public Avatar {
...
+        rollControlSpeed = value;
+    }
+
+    float getRollControlSpeed() const {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard: should be single line - when in header

**Comment by ZappoMan** [comment]
errors on build:
```
13:07:11 c:\jenkins\workspace\pr-build\label\windows\interface\src\avatar\MyAvatar.h(521): error C2065: 'rollControlSpeed' : undeclared identifier (C:\jenkins\workspace\pr-build\label\windows\interface\src\avatar\MyCharacterController.cpp) [C:\jenkins\workspace\pr-build\label\windows\build\interface\interface.vcxproj]
13:07:11   DdeFaceTracker.cpp
13:07:11 c:\jenkins\workspace\pr-build\label\windows\interface\src\avatar\MyAvatar.h(521): error C2065: 'rollControlSpeed' : undeclared identifier (C:\jenkins\workspace\pr-build\label\windows\interface\src\avatar\MyAvatar.cpp) [C:\jenkins\workspace\pr-build\label\windows\build\interface\interface.vcxproj]
13:07:12 C:\jenkins\workspace\pr-build\label\windows\interface\src\avatar/MyAvatar.h(521): error C2065: 'rollControlSpeed' : undeclared identifier (C:\jenkins\workspace\pr-build\label\windows\interface\src\avatar\AvatarActionHold.cpp) [C:\jenkins\workspace\pr-build\label\windows\build\interface\interface.vcxproj]
```

**Comment by ctrlaltdavid** [comment]
Replacement PR: https://github.com/highfidelity/hifi/pull/10603

**Pull request closed without merging**